esp-rfid.ino

```diff
@@ -182,60 +166,68 @@ void check(String card) {
           return addCard(newCard);
         }
         digitalWrite(MASTER_LED, HIGH);
-        delay(250);
+        delay(450);
         digitalWrite(MASTER_LED, LOW);
-        delay(250);
+        delay(50);
         ticks++;
       }
 
-      return success(false);
+      return success();
     }
   }
 
   for (int i = 0; i < addedCards; i++) {
     if (card == cards[i]) {
-      return success(false);
+      return success();
     }
   }
 
   digitalWrite(ERROR_LED, HIGH);
   delay(1000);
 }
 
-void hydrate() {
-  unsigned long cur = millis();
-
-  if (cur - prev >= interval) {
-    Serial.println("Reconnecting to WiFi...");
-    WiFi.disconnect();
-    WiFi.reconnect();
-    prev = cur;
-  }
-
+void digitalWriteAllLow() {
   digitalWrite(SUCCESS, LOW);
   digitalWrite(MASTER_LED, LOW);
   digitalWrite(ERROR_LED, LOW);
   digitalWrite(RELAY, LOW);
   digitalWrite(CARD_ADDED_SOUND, LOW);
+}
 
+void setup() {
+  Serial.begin(115200);
+  setupPins();
+  server.on("/open", handleOpen);
+  server.on("/card/add", handleAdd);
+  server.on("/card/list", handleList);
+  wifiSetup();
+  store.begin("rfid", false);
+  addedCards = store.getInt("count", 0);
+
+  for (int i = 0; i < addedCards; i++) {
+    String card;
+    card = store.getString(String(i).c_str(), ""); 
+    cards[i] = card;
+  }
+
+  delay(10);
+  SPI.begin();
+  mfrc522.PCD_Init();  // Initiate MFRC522
+ 
 }
 
 
 void loop() {
-  hydrate();
-  server.handleClient();
-
-  if (!mfrc522.PICC_IsNewCardPresent()) {
-    return;
-  }
-  if (!mfrc522.PICC_ReadCardSerial()) {
-    return;
+  digitalWriteAllLow();
```

Comment:
https://github.com/Lumorti/Quandoom/ whats that?

Reply:
Looks like Doom for quantum computing.
I am able to compile & run it on M1 with small modification to makefile